src/libraries/System.Text.Json/System.Text.Json.SourceGeneration.UnitTests/JsonSourceGeneratorTests.cs

```diff
@@ -51,26 +52,59 @@ public void UsePrivates()
             Compilation newCompilation = CompilationHelper.RunGenerators(compilation, out ImmutableArray<Diagnostic> generatorDiags, generator);
 
             // Make sure compilation was successful.
-            Assert.Empty(generatorDiags);
-            Assert.Empty(newCompilation.GetDiagnostics());
+            Assert.Empty(generatorDiags.Where(diag => diag.Severity == DiagnosticSeverity.Error));
+            Assert.Empty(newCompilation.GetDiagnostics().Where(diag => diag.Severity == DiagnosticSeverity.Error));
```

Comment:
We use diagnostic for users and currently we throw warnings and infos. This wasn't there currently so with the new "logs" this will not be empty anymore. We now check that diagnostics don't return error messages in compilation.